Core/Core/Router/Router.swift

```diff
@@ -18,115 +18,122 @@
 
 import UIKit
 
-public struct RouteOptions: OptionSet {
-    public let rawValue: Int
+public struct RouteOptions: Equatable {
+    public let modal: Modal?
+    public let detail: Bool
+    public let embedInNav: Bool
+    public let addDoneButton: Bool
```

Comment:
so it seems like we have 3 different places the new view ends up, and some options that apply to each.
* `.push(addDoneButton: Bool = false)`
* `.detail(embedInNav: Bool = false, addDoneButton: Bool = false)`
* `.modal(_ style: UIModalPresentationStyle? = nil, isDismissable: Bool = true, embedInNav: Bool = false, addDoneButton: Bool = false)`

I thought about using an enum for this, but that may make getting the options back out more painful. So a struct is probably fine. Maybe an enum with `case push, detail, modal(style, dismissable)` as part of the struct would be fine. 🤷‍♂

Reply:
maybe` push` shouldn't have `addDoneButton`, since it will have a back button. Any save button can't be trivially added by routing.

Reply:
maybe `detail` doesn't need an `addDoneButton` either. What would it do?